docs/pages/textarea.js

```diff
@@ -53,6 +53,265 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
+
+  return (
+    <Box width="100%">
+      <TextArea
+        id="best-practices-dont-row-height-1"
+        label="Send a message"
+        onChange={({ value }) => {
+          setValue(value);
+        }}
+        rows={1}
+        value={value}
+      />
+    </Box>
+  );
+}
+            `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <MainSection name="Accessibility">
+        <MainSection.Subsection
+          title="Comprehension"
+          description="Be sure to [provide instructions](https://www.w3.org/WAI/tutorials/forms/instructions/) to help users understand how to complete the form and use individual form controls."
+        />
+        <MainSection.Subsection
+          title="Labels"
+          description="Ensure the labels are precise and concise. Labels should only describe the text field they are associated with, and they must be visible. If you cannot use the `label` prop, ensure the alternative label's `htmlFor` attribute matches the TextArea's id. Labels are properly associated when clicking the label focuses the TextArea."
+        />
+        <MainSection.Subsection
+          title="Validation"
+          description={`
+    When providing a validation message, make sure the instructions are clear and help users complete the field. For example, "This field is required to submit". In addition, use the helper text to provide instructions to help users understand how to complete the text field or to indicate any needed input, allowed formats, timing limitations, or other pertinent information.
+    These practices give screen readers and users of assistive technologies more information about the form, helping them to fill it out.
+  `}
+        />
+        <MainSection.Subsection
+          title="Keyboard navigation"
+          description={`
+    TextArea has conventional keyboard support.
+    - Users relying on the keyboard expect to move focus to each TextArea by using the tab key or shift+tab when moving backwards
+    - Setting \`disabled\` will prevent TextArea from receiving keyboard focus or input
+  `}
+        />
+        <MainSection.Subsection
+          title="Autofocus"
+          description={`
+    TextArea intentionally lacks support for autofocus. Generally speaking,
+    autofocus interrupts normal page flow for screen readers making it an
+    anti-pattern for accessibility.
+  `}
+        />
+        <MainSection.Subsection
+          title="onSubmit"
+          description={`
+    TextArea is commonly used as an input in forms alongside submit buttons.
+    In these cases, users expect that pressing Enter or Return with the input
+    focused will submit the form.
+
+    Out of the box, TextArea doesn't expose an \`onSubmit\` handler or
+    individual key event handlers due to the complexities of handling these
+    properly. Instead, developers are encouraged to wrap TextField
+    in a \`<form>\` and attach an \`onSubmit\` handler to that \`<form>\`.
```

Comment:
The second sentence is a bit verbose and awkward; consider something like "…wrap TextField in a `<form>` with an `onSubmit` handler."